### ISSUE TITLE
Cannot save files. which is shown in journalctl as audit errors

### ISSUE THREAD

**Issue by 71walceli** [ONLYOFFICE/snap-desktopeditors#46]
From time to time, the program won't be able to save open documents. It shows a warning like this:

![Screenshot_20210416_070439](https://user-images.githubusercontent.com/26799081/115021876-2e05c800-9e82-11eb-8c5a-76acb5c541c8.png)

I'm 100% sure disk isn't full, file isn't open by any other program or "user" and file is writable. The point is that it won't save anything, being this printed by `journalctl -xef`:

```
Apr 16 07:00:31 HP-ProBook-6470b audit[14046]: AVC apparmor="DENIED" operation="mknod" profile="snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors" name="/home/apk/snap/onlyoffice-desktopeditors/67/.local-6.2.0/share/onlyoffice/desktopeditors/recover/DE__3U4M0L/XT__zCd0mO" pid=14046 comm="DesktopEditors" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
Apr 16 07:00:31 HP-ProBook-6470b audit[14046]: AVC apparmor="DENIED" operation="mkdir" profile="snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors" name="/home/apk/snap/onlyoffice-desktopeditors/67/.local-6.2.0/share/onlyoffice/desktopeditors/recover/DE__3U4M0L/XT__zCd0mO/" pid=14046 comm="DesktopEditors" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
Apr 16 07:00:31 HP-ProBook-6470b audit[14046]: AVC apparmor="DENIED" operation="mknod" profile="snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors" name="/home/apk/snap/onlyoffice-desktopeditors/67/.local-6.2.0/share/onlyoffice/desktopeditors/recover/DE__3U4M0L/params_to.xml" pid=14046 comm="DesktopEditors" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
Apr 16 07:00:31 HP-ProBook-6470b kernel: audit: type=1400 audit(1618574431.692:1244): apparmor="DENIED" operation="mknod" profile="snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors" name="/home/apk/snap/onlyoffice-desktopeditors/67/.local-6.2.0/share/onlyoffice/desktopeditors/recover/DE__3U4M0L/XT__zCd0mO" pid=14046 comm="DesktopEditors" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
Apr 16 07:00:31 HP-ProBook-6470b kernel: audit: type=1400 audit(1618574431.696:1245): apparmor="DENIED" operation="mkdir" profile="snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors" name="/home/apk/snap/onlyoffice-desktopeditors/67/.local-6.2.0/share/onlyoffice/desktopeditors/recover/DE__3U4M0L/XT__zCd0mO/" pid=14046 comm="DesktopEditors" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
Apr 16 07:00:31 HP-ProBook-6470b kernel: audit: type=1400 audit(1618574431.696:1246): apparmor="DENIED" operation="mknod" profile="snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors" name="/home/apk/snap/onlyoffice-desktopeditors/67/.local-6.2.0/share/onlyoffice/desktopeditors/recover/DE__3U4M0L/params_to.xml" pid=14046 comm="DesktopEditors" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
```

If I kill it, no autosaves will be generated, rendering editions prior to saving unrecoverable. only workaround seems to be copy-pasting data into other documents.

I think it might be related to this issue in [this other project](https://github.com/RocketChat/Rocket.Chat/issues/14562), as it only happened to `DesktopEditors` snap on my system, Every other program would still be able to fully access disk. Some similar `AppArmos` issue seems to happen to `mysql`: https://askubuntu.com/questions/916009/mysql-wont-start-because-of-apparmor. I'm unable to keep track of snap updates, so I can't confirm the errors are caused by its updates.

I appreciate any support that you can provide me.

**Comment by 71walceli** [comment]
UPDATE: I found a workaround!

It wants to write to files under `/home/apk/snap/onlyoffice-desktopeditors/67`.

In `/var/lib/snapd/apparmor/profiles/snap.onlyoffice-desktopeditors.onlyoffice-desktopeditors`, there was a line: `@{SNAP_REVISION}="68"`. I had to change it to 67, as that's the folder it was denied to write to. I'm still reading `AppArmor` and `snap` documentation.

But it's just a workaround...
